### PR TITLE
Add post-with-keys feature

### PR DESCRIPTION
**Pull request by veloce-vasta**
⬆

Also, can you check if this works?

### CODE REVIEW

File: features/post-with-keys.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+var page = function () {
+    if (document.URL.startsWith("https://scratch.mit.edu/discuss")) {
+        return "f"; // F = forums
+    } else {
+        return "e"; // E = else (not forums)
+    }
+}
+
+(function() {
+    document.onkeydown = (e) => {
```
Comment (rgantzos) [inline features/post-with-keys.js:10]: For simplicity, let's use `document.addEventListener("keydown", function() {...})` instead.

File: features/post-with-keys.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+var page = function () {
+    if (document.URL.startsWith("https://scratch.mit.edu/discuss")) {
+        return "f"; // F = forums
+    } else {
+        return "e"; // E = else (not forums)
+    }
+}
+
+(function() {
```
Comment (rgantzos) [inline features/post-with-keys.js:9]: Why is the code itself a function? It's acting like a bookmarklet, when it's in a file.

**Review by rgantzos — CHANGES_REQUESTED** [review]
^

**Comment by veloce-vasta** [comment]
Fixed

**Comment by veloce-vasta** [comment]
Fixed

File: features/post-with-keys.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+var page = function () {
+    if (document.URL.startsWith("https://scratch.mit.edu/discuss")) {
+        return "f"; // F = forums
+    } else {
+        return "e"; // E = else (not forums)
+    }
+}
+
+
+document.addEventListener(keypress, (e) => {
```
Comment (rgantzos) [inline features/post-with-keys.js:10]: "keypress" must be a string, it isn't defined as anything else.

**Review by rgantzos — CHANGES_REQUESTED** [review]

File: features/post-with-keys.js (hunk 3 above)
Comment (veloce-vasta) [reply features/post-with-keys.js:10]: Oh shoot. How did I not notice the bug? 😅

File: features/post-with-keys.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+document.addEventListener("keypress", (e) => {
+  if ((e.ctrlKey || e.metaKey) && e.key.toLowerCase() == "enter") {
+    if (document.URL.startsWith("https://scratch.mit.edu/discuss") {
```
[the remark this answers is not in the document]
Comment (veloce-vasta) [reply features/post-with-keys.js:3]: Which `if`?

[the remark this answers is not in the document]
Comment (veloce-vasta) [reply features/post-with-keys.js]: Fixed

**Comment by rgantzos** [comment]
It doesn't seem to work on Scratch 3.0 pages (project and studio comments).

**Comment by veloce-vasta** [comment]
It should work now

**Comment by veloce-vasta** [comment]
(Also, once this one is merged, is ST going to go into v21, or are you going to find more PRs to merge before the due date?)

**Comment by rgantzos** [comment]
@veloce-vasta is this still being worked on?

**Comment by veloce-vasta** [comment]
No I've been kind of busy and had some difficulties recently (in rl) so I think I'll drop this. Really sorry.

**Pull request closed without merging**